apps/src/blockly/addons/cdoTrashcan.js

```diff
@@ -2,6 +2,52 @@ import GoogleBlockly from 'blockly/core';
 import {ToolboxType} from '../constants';
 
 export default class CdoTrashcan extends GoogleBlockly.Trashcan {
+  /**
+   * @override
+   */
+  init() {
+    super.init();
+    this.workspace_.addChangeListener(this.workspaceChangeHandler.bind(this));
+  }
+
+  workspaceChangeHandler(blocklyEvent) {
+    if (blocklyEvent.type === Blockly.Events.BLOCK_DRAG) {
+      let trashcanDisplay = 'none';
+      let toolboxVisibility = 'visible';
+      const isDraggingFromFlyout_ = !!Blockly.mainBlockSpace?.currentGesture_
+        ?.flyout_;
+      if (!isDraggingFromFlyout_ && blocklyEvent.isStart) {
+        trashcanDisplay = 'block';
+        toolboxVisibility = 'hidden';
+      }
+
+      /**
+       * NodeList.forEach() is not supported on IE. Use Array.prototype.forEach.call() as a workaround.
+       * https://developer.mozilla.org/en-US/docs/Web/API/NodeList/forEach
+       */
+      Array.prototype.forEach.call(
+        // query selector for uncategorized toolbox contents
+        document.querySelectorAll('.blocklyFlyout .blocklyWorkspace'),
+        function(x) {
+          x.style.visibility = toolboxVisibility;
+        }
+      );
+      Array.prototype.forEach.call(
+        // query selector for categorized toolbox contents
+        document.querySelectorAll('.blocklyToolboxContents'),
+        function(x) {
+          x.style.visibility = toolboxVisibility;
+        }
+      );
+
+      document.querySelector('#trashcanHolder').style.display = trashcanDisplay;
+      const isDeletable = blocklyEvent.blocks.every(block =>
```

Comment:
Does this mean that if one of the blocks that's being dragged in a group of blocks is not deletable, then the trashcan won't show / the group of blocks can't be deleted?

Reply:
Yeah, that group of blocks can't be deleted. The trashcan still shows, but there's an indicator to show that you're not allowed to delete:
<img width="782" alt="Screen Shot 2021-12-01 at 2 02 25 PM" src="https://user-images.githubusercontent.com/8787187/144321376-3749c25e-dc9a-4660-89fd-f0bdaa39d252.png">